src/utils/combineReducers.js

```diff
@@ -113,12 +113,16 @@ export default function combineReducers(reducers) {
       throw sanityError;
     }
 
+    var hasChanged = false;
+
     var finalState = mapValues(finalReducers, (reducer, key) => {
-      var newState = reducer(state[key], action);
+      var originalState = state[key];
+      var newState = reducer(originalState, action);
```

Comment:
I think it's a little confusing now that we have many `*State` variables, almost all of which refer to slices of state, but some refer to the state object itself.

Let's rename `originalState` and `newState` to `previousStateByKey` and `nextStateByKey`?